### ISSUE TITLE
about the test mode

### ISSUE THREAD

**Issue by XDUSPONGE** [BindsNET/bindsnet#372]
First, thanks very much for your shared code
When I run the code as ptyhon ethmnist.py 
I find for the dataloader the training_data is the same with the test_data
So, if there is anything wrong?
And my env is 
pytorch 1.4 
cuda10.1

**Comment by Hananel-Hazan** [comment]
Hi, thank you for using BindNET.

You need to create another dataloader  with the option of  `train=False`, and then use this object in your testing loop. 

For example:

```
# Load MNIST data.
test_dataset = MNIST(
    PoissonEncoder(time=time, dt=dt),
    None,
    root=os.path.join("..", "..", "data", "MNIST"),
    download=True,
    train=False,
    transform=transforms.Compose(
        [transforms.ToTensor(), transforms.Lambda(lambda x: x * intensity)]
    ),
)
```

**Comment by XDUSPONGE** [comment]
hello,  thanks for your reply, as the paper says, 
```
After trianing is done, we set the learning rate to zero, fix each neuron's spiking threshould, and assign a class to each neuron, based on its highest response to the ten classes of digits over one presentation of the training seet
```
so the whole pipeline I understood is 
```
train_dataloader = 
test_dataloader = 
for epoch in range(n_epochs):
	# training ....
	for step, batch in enumerate(tqdm(train_dataloader)):
		if uptate_inter:
			activity_pred = ..
			accuracy = ..
			assign_labels ..

		network.run(inputs)
	# testing ....
	for step, batch in enumerate(tqdom(test_dataloader)):
		if uptate_inter:
			activity_pred = ...
			accuracy = ...
			#test does not assign label
		network.run(inputs)
```
so my question is 
how to keep the  set the learning rate to zero, fix each neuron's spiking threshould afterting the training is done,  the pipeline I do did not to these two things

**Comment by Hananel-Hazan** [comment]
Please use: `network.train(mode=False)` before starting the testing process. That should disable the synapses modulation.
When you want to go back and train the network use: `network.train(mode=True)`.
